### PR TITLE
Blac 19 registration

### PR DESCRIPTION
**Pull request by Obibus**

### CODE REVIEW

File: src/test/java/base/AbstractBaseTest.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
 
-abstract public class AbstractBaseTest {
+abstract public class AbstractBaseTest{
```
Comment (NataLitovets) [inline src/test/java/base/AbstractBaseTest.java]: Add space between AbstractBaseTest{

File: src/test/java/elements/HomeElements.java [hunk 2]
```diff
@@ -8,4 +8,8 @@ public HomeElements(WebDriver driver) {
         super(driver);
     }
 
+    public static final String ACCOUNT_BUTTON = "//div//button[@data-test-id='account-button']";
```
Comment (NataLitovets) [inline src/test/java/elements/HomeElements.java]: you can remove "//div" in XPath and rename to ACCOUNT_ICON_BUTTON

File: src/test/java/elements/HomeElements.java [hunk 3]
```diff
@@ -8,4 +8,8 @@ public HomeElements(WebDriver driver) {
...
+    public static final String REGISTER_HERE_BUTTON_ON_ACCOUNT_MENU = "//div//a[@data-test-id='register-button']";
```
Comment (NataLitovets) [inline src/test/java/elements/HomeElements.java]: please. separate account menu elements by adding a comment like "//account menu elements" and "//homepage elements"

you can remove "//div" in XPath and rename REGISTER_HERE_BUTTON

File: src/test/java/elements/RegisterElements.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import base.AbstractBasePage;
+import org.openqa.selenium.WebDriver;
+
+public class RegisterElements extends AbstractBasePage {
+    public RegisterElements(WebDriver driver) {
+        super(driver);
+    }
+
+    public static final String FIRST_NAME_FIELD = "//div//input[@data-test-id='first-name']";
```
Comment (NataLitovets) [inline src/test/java/elements/RegisterElements.java]: if you are using data-test-id you can call element by using only it (remove //div here and everywhere where you use data-test-id)

File: src/test/java/elements/RegisterElements.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    public static final String REGISTER_BUTTON = "//div//button[@data-test-id='register']";
+    public static final String LOGIN_ALREADY_IN_USE_TEXT = "//p[@data-test-id='registration-form-error-content']";
+    public static final String CREATE_ACCOUNT_BUTTON_ON_TOP = "//a[@data-test-id='register-action-button']";
+    public static final String TERMS_AND_CONDITIONS = "//a[text()='Terms and Conditions']";
+    public static final String PRIVACY_POLICY = "//div[@class='text-xs normal-case']/p//a[text()='Privacy Policy']";
+    public static final String GENERAL_TERMS_AND_CONDITIONS = "//h2[@class='font-bold']/strong";
+    public static final String FORGOTTEN_YOUR_PASSWORD = "//button[@data-test-id='forgotten-password-link']";
+    public static final String SUBMIT_IN_FORGOTTEN_PASSWORD = "//button[@data-test-id='submit-password-reset-btn']";
+    public static final String EMAIL_FIELD_IN_FORGOTTEN_PASSWORD = "//input[@data-test-id='email-input']";
+    public static final String EMAIL_FIELD_ON_LOGIN_PAGE = "//input[@data-test-id='auth-field-email']";
```
Comment (NataLitovets) [inline src/test/java/elements/RegisterElements.java:25]: please. separate create account elements by adding a comment like "//create account elements" and "//login elements"

File: src/test/java/pages/HomePage.java [hunk 6]
```diff
@@ -29,6 +29,25 @@ public HomePage clickClassicsWomenShoes() {
...
+    public HomePage clickLoginButtonOnAccountMenu() {
+        waitUntilElementToBeClickableByXpath(LOGIN_BUTTON_ON_ACCOUNT_MENU).click();
+        return this;
+    }
+
+    public HomePage clickReturnHomePagePumaIcon() {
+        waitUntilElementToBeClickableByXpath(RETURN_HOME_PAGE_PUMA_ICON).click();
+        return this;
+    }
+
```
Comment (NataLitovets) [inline src/test/java/pages/HomePage.java]: remove empty line

File: src/test/java/elements/RegisterElements.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package elements;
+
+import base.AbstractBasePage;
+import org.openqa.selenium.WebDriver;
+
+public class RegisterElements extends AbstractBasePage {
```
Comment (NataLitovets) [inline src/test/java/elements/RegisterElements.java:6]: Add empty line after class declaration

File: src/test/java/pages/RegisterPage.java [hunk 8]
```diff
@@ -0,0 +1,91 @@
+package pages;
+
+import elements.RegisterElements;
+import org.openqa.selenium.WebDriver;
+
+public class RegisterPage extends RegisterElements {
```
Comment (NataLitovets) [inline src/test/java/pages/RegisterPage.java:6]: Add empty line after class declaration

File: src/test/java/pages/RegisterPage.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+    public RegisterPage fillLoginPassword(String LoginPassword) {
+        waitUntilElementToBeClickableByXpath(PASSWORD_FIELD_ON_LOGIN_PAGE).sendKeys(LoginPassword);
+        return this;
+    }
+
+    public RegisterPage clickLoginButtonOnLoginPage() {
+        waitUntilElementToBeClickableByXpath(LOGIN_BUTTON_ON_LOGIN_PAGE).click();
+        return this;
+    }
+
```
Comment (NataLitovets) [inline src/test/java/pages/RegisterPage.java]: remove empty line

File: src/test/java/tests/RegistrationTest.java [hunk 10]
```diff
@@ -0,0 +1,170 @@
+package tests;
+
+import base.AbstractBaseTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+import pages.HomePage;
+import pages.RegisterPage;
+
+public class RegistrationTest extends AbstractBaseTest {
```
Comment (NataLitovets) [inline src/test/java/tests/RegistrationTest.java:10]: Add empty line after class declaration

File: src/test/java/tests/RegistrationTest.java [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+        registerPage.fillLastName(userLastName);
+        registerPage.fillEmail(userEmail);
+        registerPage.fillPassword(userPassword);
+
+        if (shouldClickRegister == true) {
+            registerPage.clickRegister();
+            registerPage.waitLoginAlreadyInUse();
+        }
+
+        pageSource = driver.getPageSource();
```
Comment (NataLitovets) [inline src/test/java/tests/RegistrationTest.java]: Move this method in AbstractBaseTest

File: src/test/java/tests/RegistrationTest.java [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+        softAssert.assertTrue(pageSourceTermsAndConditions.contains("General Terms and Conditions"), "PUMA.com General Terms and Conditions");
+
+        homePage.clickReturnHomePagePumaIcon();
+        homePage.clickAccountButton();
+        homePage.clickRegisterHereButton();
+        registerPage.clickPrivacyPolicy();
+
+        pageSourcePrivacyPolicy = driver.getPageSource();
+        softAssert.assertTrue(pageSourcePrivacyPolicy.contains("Privacy Policy"), "Privacy Policy");
+    }
```
Comment (NataLitovets) [inline src/test/java/tests/RegistrationTest.java:121]: You forgot assertAll

File: src/test/java/tests/RegistrationTest.java [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+        pageSource = driver.getPageSource();
+        if (userEmail.equals("maksbidzura@gmail.com")) {
+            softAssert.assertTrue(pageSource.contains("My account"), "My account");
+        } else {
+            softAssert.assertTrue(pageSource.contains("You need to have a valid email"), "You need to have a valid email");
+        }
+
+        softAssert.assertAll();
+    }
+
```
Comment (NataLitovets) [inline src/test/java/tests/RegistrationTest.java]: remove empty line

File: src/test/java/base/AbstractBaseTest.java (hunk 1 above)
Comment (Obibus) [reply src/test/java/base/AbstractBaseTest.java]: corrected

File: src/test/java/elements/HomeElements.java (hunk 2 above)
Comment (Obibus) [reply src/test/java/elements/HomeElements.java]: corrected

File: src/test/java/elements/RegisterElements.java (hunk 4 above)
Comment (Obibus) [reply src/test/java/elements/RegisterElements.java]: corrected

File: src/test/java/tests/RegistrationTest.java (hunk 10 above)
Comment (Obibus) [reply src/test/java/tests/RegistrationTest.java:10]: corrected

File: src/test/java/pages/RegisterPage.java (hunk 9 above)
Comment (Obibus) [reply src/test/java/pages/RegisterPage.java]: corrected

File: src/test/java/tests/RegistrationTest.java (hunk 11 above)
Comment (Obibus) [reply src/test/java/tests/RegistrationTest.java]: corrected

File: src/test/java/elements/HomeElements.java (hunk 3 above)
Comment (Obibus) [reply src/test/java/elements/HomeElements.java]: corrected

File: src/test/java/elements/RegisterElements.java (hunk 5 above)
Comment (Obibus) [reply src/test/java/elements/RegisterElements.java:25]: corrected

File: src/test/java/pages/RegisterPage.java (hunk 8 above)
Comment (Obibus) [reply src/test/java/pages/RegisterPage.java:6]: corrected

File: src/test/java/elements/RegisterElements.java (hunk 7 above)
Comment (Obibus) [reply src/test/java/elements/RegisterElements.java:6]: corrected

File: src/test/java/pages/HomePage.java (hunk 6 above)
Comment (Obibus) [reply src/test/java/pages/HomePage.java]: corrected

File: src/test/java/tests/RegistrationTest.java (hunk 12 above)
Comment (Obibus) [reply src/test/java/tests/RegistrationTest.java:121]: corrected

File: src/test/java/tests/RegistrationTest.java (hunk 13 above)
Comment (Obibus) [reply src/test/java/tests/RegistrationTest.java]: corrected

**Review by NataLitovets — APPROVED** [review]

**Review by DashaYab — APPROVED** [review]

**Pull request merged**